web/client/components/TOC/DefaultLayer.jsx

```diff
@@ -118,12 +119,15 @@ class DefaultLayer extends React.Component {
             <div key="legend" position="collapsible" className="collapsible-toc">
                 <Grid fluid>
                     {this.props.showFullTitleOnExpand ? <Row><Col xs={12} className="toc-full-title">{this.getTitle(this.props.node)}</Col></Row> : null}
-                    {this.props.activateLegendTool ?
+                    {this.props.activateLegendTool && this.props.node.type === 'wms' &&
                         <Row>
                             <Col xs={12}>
                                 <WMSLegend node={this.props.node} currentZoomLvl={this.props.currentZoomLvl} scales={this.props.scales} language={this.props.language} {...this.props.legendOptions} />
                             </Col>
-                        </Row> : null}
+                        </Row>}
+                    {this.props.activateLegendTool && ['wfs', 'vector'].includes(this.props.node.type) &&
+                        <WFSLegend rules={this.props.node.style.body.rules}/>
```

Comment:
Instead to pass the rules array we should pass the style object (`this.props.node.style`) so the Legend component could:
- if the format is geostyler use directly the style rules
- if the style is not supported don't render the legend

In the future we could extend the possibility to convert from different format to geostyler style inside the Legend component